app/auth/web.py

```diff
@@ -56,6 +57,11 @@
 else:
     COOKIE_DOMAIN = None
 
+if os.environ.get('QUART_DEBUG', "0") != "1":
+    COOKIE_SECURE = {'secure': True, 'httponly': True}  # keep httponly when UI is also adapted
```

Comment:
I think this will currently not work for a minikube-deployment in which the gateway is NOT swapped by telepresence. In this case `QUART_DEBUG` will not be set, therefore `secure` will be `True`, however the UI is still not served through https.

I suggest to keep `httponly` always set (less risk of introducing bugs in a dev setting where js reads from the cookie) and the `COOKIE_SECURE` env variable should be set in `deployment.yaml` inside the chart templates (there is already a `gateway.protocol`) defined that you can use: https://github.com/SwissDataScienceCenter/renku-gateway/blob/master/helm-chart/renku-gateway/templates/_helpers.tpl

Reply:
I set the httponly for both cases and changed the condition to look at the HOST_NAME, which is injected with the gateway.protocol and then should reflect the fact that we are using https or not.

But it seems the UI is not currently sending the X-Requested-With header

Reply:
Yes, the UI needs to be updated too to match the changes in the gateway.